### DISCUSSION TITLE
User Interface Modification - Add an option to show projects as a list...

### DISCUSSION THREAD

**Discussion by LukasKM** [openkfw/TruBudget#1040] [Ideas]
When there is a higher number of projects, the current "card view" becomes difficult to navigate in, despite the existing search function. 
It would be great if users have an option to switch from "card view" to a "list view" that would allow to show more projects on the screen. 

This could be combined with: 

1. Filtering of closed projects (e.g. simle switch include/exclude closed projects).
2. Sorting of projects according to the project status, i.e. the open projects on the top, closed on the bottom.
3. Grouping of projects (in a list view) by their tags, e.g. if there are several projects with the same tags, they would be shown within the same goup in that list.

**Comment by Stezido** [comment]
Hi Lukas,
grouping of projects via tags or name is already implemented: https://trubudget.net/docs/user-guide/projects/project#project-tags
I agree that this solution needs to be improved though. 
Implementing a list view with filtering and sorting options sounds like the best way to handle a high number of projects.
We will considering this feature in our next sprint!

**Comment by schoel-bis** [comment]
There are a few issues also with the current card view of the projects:

* Longer project names (> 20~25 characters) are ellipsized and there appears to be no way in the card view to see the full project name. Not sure what would be a good solution to this (auto-scale the text? Reveal on hover?).
* It is hard to distinguish open and closed projects. In addition to what Lukas has proposed above I would suggest to also provide some clear visual indication of the status, e.g. 60% opacity until hovered, greyish typeface, greyish background, some sort of badge, …
* It could be useful to have an (or several) alternative set of pictures for the cards, maybe something more abstract or the option to have the picture algorithmically generated, e.g. based on the project's UUID. That would still help with distinguishing projects visually but would make it easier to pick a picture, in particular when auto-generating projects.
